### PR TITLE
Create submit survey page

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Issue #:*
https://github.com/bcgov/met-public/issues/286

*Description of changes:*

- Create submit survey page
- Use Form.io form viewer component to display created forms
- Add submit survey route
- Add buttons on engagement view page to navigate to survey submit page
- Update the engagement banner to newer design

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/resources/submission.py [hunk 1]
```diff
@@ -61,12 +61,8 @@ class Submissions(Resource):
     def post():
         """Create a new submission."""
         try:
-            user_id = TokenInfo.get_id()
             requestjson = request.get_json()
             schema = SubmissionSchema().load(requestjson)
-            schema['user_id'] = user_id
```
Comment (tiagograf-AOT) [inline met-api/src/met_api/resources/submission.py:67]: I'm guessing we implement this once we have some sort of authentication for the public

**Review by tiagograf-AOT — APPROVED** [review]

File: met-web/src/components/engagement/view/EngagementBanner.tsx [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
+import { Banner } from '../banner/Banner';
+import { useNavigate } from 'react-router-dom';
 
 export const EngagementBanner = () => {
+    const navigate = useNavigate();
     const { engagementLoading, savedEngagement } = useContext(ActionContext);
 
-    const imageExists = !!savedEngagement.banner_url;
-
+    const surveyId = savedEngagement.surveys[0]?.id || '';
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/view/EngagementBanner.tsx:11]: I don't love that we are prepping for multiple surveys but there's stuff that is still fixed for only one, its just a comment no change for now

Comment (jadmsaadaot) [reply met-web/src/components/engagement/view/EngagementBanner.tsx:11]: I understand, but it wasn't really a decision here. I noticed that this  db.relationship('Survey', backref='engagement', cascade='all, delete') returns an array. So I continued handling it like that and thought to keep it open for multiple surveys.

File: met-api/src/met_api/resources/submission.py (hunk 1 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/resources/submission.py:67]: true, I don't think though the user_id will come from a jwt. I think flow is, submit email, receive survey link in email that has a valid period. Then submit survey.

File: met-api/src/met_api/models/survey.py [hunk 3]
```diff
@@ -21,18 +21,18 @@ class Survey(db.Model):  # pylint: disable=too-few-public-methods
...
     updated_by = db.Column(db.String(50))
+    engagement_id = db.Column(db.Integer, ForeignKey('engagement.id', ondelete='CASCADE'))
 
     @classmethod
     def get_survey(cls, survey_id) -> SurveySchema:
         """Get a survey."""
         survey_schema = SurveySchema()
-        data = db.session.query(Survey).join(Engagement, isouter=True)\
-            .join(EngagementStatus, isouter=True).filter_by(id=survey_id).first()
+        data = db.session.query(Survey).filter_by(id=survey_id).join(Engagement, isouter=True)\
```
[the remark this answers is not in the document]
Comment (jadmsaadaot) [reply met-api/src/met_api/models/survey.py:34]: agreed

**Review by djnunez-aot — APPROVED** [review]

**Pull request merged**